### PR TITLE
19 tracking analytics

### PR DESCRIPTION
**Pull request by Edhoru**
This pull request includes the changes to track screens in the sdk

IMPORTANT: don't merge before: https://github.com/schedjoules/ios-public-calendars-sdk/pull/23

### CODE REVIEW

File: Demo/AppDelegate.swift [hunk 1]
```diff
@@ -18,12 +19,14 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
         
         // Initialize the calendar store
-        let calendarVC = CalendarStoreViewController(apiKey: "0443a55244bb2b6224fd48e0416f0d9c", title: "Featured")
+        let calendarVC = CalendarStoreViewController(title: "Featured")
         
         // Show the calendar store
         window?.rootViewController = calendarVC
         window?.makeKeyAndVisible()
 
+        AnalyticsTracker.shared().launch()
```
Comment (dmfs) [inline Demo/AppDelegate.swift]: What happens if a partner doesn't include this call into his code? Why don't you do this inside the ctor of `CalendarStoreViewController`?

File: SDK/CalendarStoreViewController.swift [hunk 2]
```diff
@@ -63,9 +71,9 @@ public final class CalendarStoreViewController: UITabBarController {
      - parameter pageIdentifier: The page identifier for the the home page.
      - parameter title: The title for the `navigtaion bar` in the home page.
      */
-    public init(apiClient: Api, pageIdentifier: String?, title: String?) {
+    public init(pageIdentifier: String?, title: String?) {
         // Initialization
-        self.apiClient = apiClient
+        self.apiClient = SchedJoulesApi(accessToken: apiKey)
```
Comment (dmfs) [inline SDK/CalendarStoreViewController.swift]: Not a fan of this change. We expect partners to keep their API key secret. If they pass it to the constructor they can decide themselves where and how to store it. If they choose to store it in the `Info.plist` from where any other SDK they use can read it it's up to them, but we shouldn't make it mandatory.
That's also not how other services do it, see how you have to provide the APK Key to Google Services SDK: https://developers.google.com/maps/documentation/ios-sdk/get-api-key

File: SDK/New Group/Analytics/AnalyticsTracker.swift [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+    private func finishProcessing() {
+        self.isProcessing = false
+        self.loopHits()
+    }
+    
+    private func uploadHits() {
+        DispatchQueue.global(qos: .background).async {
+            
+            guard let hits = UserDefaults.standard.trackingHits.splitBy(size: self.sizeForUpload).first else {
+                print("all hits uploaded")
```
[the remark this answers is not in the document]
Comment (Edhoru) [reply SDK/New Group/Analytics/AnalyticsTracker.swift]: I added a logger that only works on Debug, to test it fully I need first to create the framework and import it in a different project, either way now we can silence prints with just one line

https://github.com/schedjoules/ios-public-calendars-sdk/pull/25/commits/6e5d206490da7173bc78921fb2c4582f2a33ca69

File: Demo/AppDelegate.swift (hunk 1 above)
Comment (Edhoru) [reply Demo/AppDelegate.swift]: Moved https://github.com/schedjoules/ios-public-calendars-sdk/pull/25/commits/a5c4b53717467daaef473c6d66de493d650b954d

**Pull request merged**